helix-core/src/main/java/org/apache/helix/controller/stages/ResourceComputationStage.java

```diff
@@ -185,4 +232,21 @@ private void addPartition(String partition, String resourceName, Map<String, Res
     resource.addPartition(partition);
 
   }
+
+  private void addResourceConfigToResourceMap(String resourceName, ResourceConfig resourceConfig,
```

Comment:
So it seems that all `WorkflowConfigs` and `JobConfigs` are being added to both `resourceMap` and `resourceToRebalance`. 

If my understanding of the pipeline is up-to-date, the reason we have two different maps is for the resource pipeline to filter out TASK resources and save them into `resourceToRebalance`.
So 
- `resourceMap`: all resources 
- `resourceToRebalance`: all resources - TASK resources

Now, for the task pipeline, do we need to keep both maps? It seems that you're doing the filtering by populating things from `ResourceConfigs` only + `currentStates` to `resourceToRebalance`. With that said, what's the point of populating/keeping `resourceMap`? Does `resourceMap` still contain ALL resources for the task pipeline as well?

Reply:
This PR doesn't change any of the map populating behavior: if a workflow/job was added to both maps in the past (based on IS), now they are added to both maps based on configs. With that said, your assessment on `resourceToRebalance` isn't entirely accurate: when the pipeline is for tasks, `resourceToRebalance` contains all the task resources as well. 

The reason why we have both maps is because of the usages in other stages. For example, in `CurrentStateComputationStage`, task resources need to be present in both maps. There's definitely room for improvement in this aspect, but that would perhaps be in another PR.

Reply:
> With that said, your assessment on `resourceToRebalance` isn't entirely accurate: when the pipeline is for tasks, `resourceToRebalance` contains all the task resources as well.

What's not accurate? Do you mean to say that for the resource pipeline, resourceToRebalance contains task resources? For the task pipeline, should resourceMap contain all resources and resourceToRebalance contain only task resources?

Reply:
I see that I had a bit misunderstanding on your original message: you're right that in the resource pipeline, `resourceToRebalance` doesn't contain task resources; in the task pipeline, `resourceToRebalance` only contains task resources. This behavior remains the same after IS removal (`resourceMap` contains all resources).